spec/consumer_resource_spec.rb

```diff
@@ -1519,20 +1519,23 @@ def verify_installed_pids(consumer, product_ids)
     unexpected_exceptions = []
     expected_exceptions = []
     threads = []
+    lock = Mutex.new
     total_threads.times do
       t = Thread.new do
-        begin
-          @user1.unregister(consumer['uuid'])
-        rescue RestClient::ResourceNotFound => rnf
-          # Expected and OK.
-          expected_exceptions << rnf
-        rescue RestClient::Gone => gone
-          expected_exceptions << gone
-        rescue Exception => e
-          # Unexpected - report it.
-          unexpected_exceptions << e
+        lock.synchronize do
```

Comment:
This test (and the other ones you have edited in the PR), seems to want to run the threads in parallel by design, in order to test what happens on the server side when concurrent requests happen. If you use a lock these blocks of code, you essentially only let one thread execute this at a time, which defeats the purpose of the test.

I would focus on finding out what changed in the infrastructure first (some examples: perhaps Ruby version/interpreter changed that now uses true multithreading (see regarding the Global Interpreter Lock [here](https://www.toptal.com/ruby/ruby-concurrency-and-parallelism-a-practical-primer), while before it wasn't? perhaps there's difference in how containers work now that our infrastructure moved to fedora 30 (maybe the docker/podman version used has different multithreading support)? 

If we find out that indeed we were not actually running these tests in parallel, we might actually need to fix the way the server responds when _truly concurrent_ requests like these come in.